ui/app/AppLayouts/Wallet/controls/SavedAddressesDelegate.qml

```diff
@@ -114,10 +130,32 @@ StatusListItem {
                                      edit: true,
                                      address: editDeleteMenu.contactAddress,
                                      name: editDeleteMenu.contactName,
-                                     favourite: editDeleteMenu.storeFavourite
+                                     favourite: editDeleteMenu.storeFavourite,
+                                     chainShortNames: editDeleteMenu.contactChainShortNames,
+                                     ens: editDeleteMenu.contactEns
                                  })
             }
         }
+        StatusAction {
+            text: qsTr("Copy")
+            objectName: "copySavedAddressAction"
+            assetSettings.name: "copy"
+            onTriggered: {
+                if (d.visibleAddress)
+                    store.copyToClipboard(d.visibleAddress)
+                else
+                    store.copyToClipboard(root.ens)
+            }
+        }
+        StatusMenuSeparator { }
+        StatusAction {
+            text: qsTr("View on Etherscan")
+            objectName: "viewOnEtherscanAction"
+            assetSettings.name: "external"
+            onTriggered: {
+                Qt.openUrlExternally("https://etherscan.io/address/%1".arg(d.visibleAddress ? d.visibleAddress : root.ens))
```

Comment:
I am always wondering have we decided to open links in the user's browser by default or go through `Global.openLink` and ask him whether to open "in app" or externally?

Reply:
Use `Global.openLink()` for consistency (eventhough we now default to opening in an external browser anyway; we might want to reconsider that in the future)

Reply:
Still valid